tests/js/server/dump/dump-test.inc

```diff
@@ -51,6 +51,34 @@
     return c;
   }
 
+  function waitForCollInSync(collName) {
+    let res;
+    let numSecs = 0.5;
+    let collDistribution = {};
+    let collDistributionPlan = {};
+    let collDistributionCurrent = {};
+    while (true) {
+      const dbName = db._name();
+      const res = db._connection.GET("/_db/" + dbName + "/_admin/cluster/shardDistribution?returnAsJSON=true");
+      const colls = res["results"];
+      for (const key in colls) {
+        if (key.includes(collName)) {
+          collDistribution = colls[key];
+        }
+      }
+      assertFalse(_.isEmpty(collDistribution));
+      collDistributionPlan = collDistribution["Plan"];
+      collDistributionCurrent = collDistribution["Current"];
+      assertFalse(_.isEmpty(collDistributionPlan));
+      assertFalse(_.isEmpty(collDistributionCurrent));
+      if (_.isEqual(collDistributionPlan, collDistributionCurrent) || numSecs >= 16) {
```

Comment:
if we reach `numSecs >= 16`, should we continue the test anyway? it will very likely fail then.
Why not assert here that numSecs is < 16, and if the assertion fails, use a descriptive error message so that developers can see that the shards did not get in sync. This will be better than random hot_backup test failures down the line that are hard to understand and debug.